sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3390,13 +3390,13 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
-  val REPLACE_NULLIF_USING_WITH_EXPR =
-    buildConf("spark.databricks.sql.replaceNullIfUsingWithExpr")
+  val ALWAYS_INLINE_COMMON_EXPR_IN_WITH =
+    buildConf("spark.sql.alwaysInlineCommonExprInWith")
       .internal()
-      .doc("When true, NullIf expressions are rewritten using With expressions to avoid " +
-        "expression duplication.")
+      .doc("When true, always inline common expressions inside WITH expression which may lead " +
```

Comment:
when this config is true, we don't inline the common expressions IN the with expressions, rather we just inline the common expressions? should we remove "inside WITH expression"?